lib/metadata/event-debouncer.js

```diff
@@ -0,0 +1,114 @@
+"use strict";
+
+var utils = require('../utils');
+
+/**
+ * Debounce protocol events by acting on those events with a sliding delay.
+ * @param {Number} delay
+ * @ignore
+ * @constructor
+ */
+function EventDebouncer(delay) {
+  this._delay = delay;
+  this._queue = null;
+  this._timeout = null;
+}
+
+/**
+ * Adds a new event to the queue and moves the delay.
+ * @param {{ handler: Function, all: boolean|undefined, keyspace: String|undefined,
+ *  cqlObject: String|undefined, callback: Function|undefined }} event
+ */
+EventDebouncer.prototype.eventReceived = function (event) {
+  event.callback = event.callback || utils.noop;
+  this._queue = this._queue || { callbacks: [], keyspaces: {} };
+  //TODO: Warn when size is too large
```

Comment:
> After NODEJS-106, STATUS_CHANGE events have no significant effect in the driver.

That's a good point, in fact the same applies to the java driver as well now, so that might be a good consideration there as well.

> I think sharing a "single" debouncer helps setting a hierarchy: if the hosts + keyspaces are metadata is refreshed, there is no point in refreshing a single keyspace / table.

I agree, makes sense and it is a lot easier to conceptualize vs. having separate debouncers on their own timers.